<a href="https://colab.research.google.com/github/Lizzy-TM/sistema-de-registro-2/blob/main/Curso_de_Estad%C3%ADstica_con_Python_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

***
# <font color=#56B4E9 size=10>CURSO DE ESTADÍSTICA</font>
***

<font color=#56B4E9 size = 5>*Preparando el ambiente*</font>

# <font color=#56B4E9>1 CONOCIENDO LOS DATOS</font>
***

## <font color=#56B4E9>1.1 Dataset del proyecto</font>
***

### Muestra de domicilios Colombia - 2018

Las investigaciones por muestras de domicilios realizadas anualmente, buscan encontrar característcias generales de la población, de educación, trabajo, rendimiento y otras, de acuerdo con las necesidades de información del país, tales como las características de migración, fertilidad, casamientos, salud, nutrición, entre otros temas. Estas muestras al pasar de los años consistuyen una herramienta importante para la formulación, validación y evaluación de políticas dirigidas al desarrollo socioeconómico y la mejora de las condiciones de vida en Colombia.


### Datos

Los datos fueron creados  de manera didáctica para este curso.

### Variables utilizadas

> ### Ingreso
> ***

Ingresos mensuales (en miles de pesos) del trabajo principal para personas de 10 años o más.

> ### Edad
> ***

Edad del entrevistado en la fecha de referencia en años.

> ### Altura
> ***

Altura del entrevistado em metros.

> ### Ciudad
> ***

Código de referência a 27 ciudades analizadas.


> ### Sexo
> ***

|Código|Descripción|
|---|---|
|0|Masculino|
|1|Femenino|

> ### Años de Estudio
> ***

|Código|Descripción|
|---|---|
|1|Sin estudios y menos de 1 año|
|2|1 año|
|3|2 años|
|4|3 años|
|5|4 años|
|6|5 años|
|7|6 años|
|8|7 años|
|9|8 años|
|10|9 años|
|11|10 años|
|12|11 años|
|13|12 años|
|14|13 años|
|15|14 años|
|16|15 años o más|
|17|No se sabe|
||No aplica|

> ### Color
> ***

|Código|Descripción|
|---|---|
|0|Indio|
|2|Blanco|
|4|Negro|
|6|Amarillo|
|8|Moreno|
|9|Sin declarar|

> ### Tratamiento a los datos
***

> Algunos de los tratamientos de datos más frecuentes son:
> 1. Eliminar las observaciones (líneas) con entradas de datos inválidos;
> 2. Eliminar observaciones donde hay datos perdidos (missing data);
> 3. Filtros própios de la investigación, por ejemplo: considerar solo las encuestas realizadas a la cabeza de familia (responsable por el domicilio).

### Importando pandas y leyendo el dataset del projecto

https://pandas.pydata.org/

In [27]:
import pandas as pd

In [28]:
datos = pd.read_csv("/content/datos.csv")

In [29]:
import pandas as pd

In [30]:
datos=pd.read_csv('/content/datos.csv')

In [31]:
datos.head()

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
0      11     0    23      8               12      800  1.603808
1      11     1    23      2               12     1150  1.739790
2      11     1    35      8               15      880  1.760444
3      11     0    46      2                6     3500  1.783158
4      11     1    47      8                9      150  1.690631

In [8]:
datos.sample(10)

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
40955      32     0    38      8               16     3000  1.682531
41498      33     0    33      4                7     2000  1.737838
13779      22     0    47      4               12      500  1.689740
16989      23     0    29      8               12     2364  1.708486
1667       12     0    30      2               12     1000  1.690361
19271      25     0    26      2                7      788  1.711419
10923      17     1    31      8                7      400  1.720774
62172      43     0    33      2                1     1500  1.629820
4305       13     0    33      8               12      788  1.653403
11699      21     0    42      8                4     1000  1.742716

In [9]:
datos.sample(10)

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
70942      51     0    26      8               12     1500  1.814084
4004       13     0    46      4                9      976  1.605522
22538      26     0    48      2                9     6000  1.662175
20570      26     1    68      8                3        0  1.667206
62162      43     0    43      2               12     1600  1.691981
43537      33     0    33      2               16     7300  1.684840
50137      35     0    41      2               12     1300  1.704490
40135      32     0    45      8                6      908  1.702046
19779      25     0    36      8               12       33  1.604747
55697      41     1    24      8               12     1800  1.650094

## <font color=#56B4E9>1.2 Tipos de datos</font>
***

### Clasificación de una variable
<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img001.png' width='70%'>

### Variables cualitativas ordinales

► Variables que pueden ser ordenadas o que responen algun tipo de jerarquía

In [10]:
sorted(datos['Años de Estudio'].unique())

[np.int64(1),
 np.int64(2),
 np.int64(3),
 np.int64(4),
 np.int64(5),
 np.int64(6),
 np.int64(7),
 np.int64(8),
 np.int64(9),
 np.int64(10),
 np.int64(11),
 np.int64(12),
 np.int64(13),
 np.int64(14),
 np.int64(15),
 np.int64(16),
 np.int64(17)]

In [11]:
datos['Años de Estudio'].unique()

array([12, 15,  6,  9, 14, 16, 10,  1,  5,  7,  3,  2,  4,  8, 11, 17, 13])

In [12]:
sorted(datos['Años de Estudio'].unique())

[np.int64(1),
 np.int64(2),
 np.int64(3),
 np.int64(4),
 np.int64(5),
 np.int64(6),
 np.int64(7),
 np.int64(8),
 np.int64(9),
 np.int64(10),
 np.int64(11),
 np.int64(12),
 np.int64(13),
 np.int64(14),
 np.int64(15),
 np.int64(16),
 np.int64(17)]

### Variables cualitativas nominales

► Variables que no pueden ser ordenadas ni responden alguna jerarquía.

In [13]:
sorted(datos['Sexo'].unique())

[np.int64(0), np.int64(1)]

In [14]:
sorted(datos['Color'].unique())

[np.int64(0), np.int64(2), np.int64(4), np.int64(6), np.int64(8)]

In [15]:
sorted(datos['Color'].unique())

[np.int64(0), np.int64(2), np.int64(4), np.int64(6), np.int64(8)]

In [16]:
sorted(datos['Ciudad'].unique())

[np.int64(11),
 np.int64(12),
 np.int64(13),
 np.int64(14),
 np.int64(15),
 np.int64(16),
 np.int64(17),
 np.int64(21),
 np.int64(22),
 np.int64(23),
 np.int64(24),
 np.int64(25),
 np.int64(26),
 np.int64(27),
 np.int64(28),
 np.int64(29),
 np.int64(31),
 np.int64(32),
 np.int64(33),
 np.int64(35),
 np.int64(41),
 np.int64(42),
 np.int64(43),
 np.int64(50),
 np.int64(51),
 np.int64(52),
 np.int64(53)]

### Variables cuantitativas discretas

► Variables que representan un conteo donde los valores posibles forman un conjunto finito o numerable.


In [17]:
datos.Edad.max()

99

In [18]:
datos.Edad.unique()

array([23, 35, 46, 47, 34, 57, 60, 50, 26, 49, 52, 38, 45, 43, 53, 62, 69,
       41, 25, 33, 44, 65, 32, 30, 64, 31, 63, 40, 28, 37, 39, 36, 73, 56,
       27, 42, 51, 29, 24, 55, 67, 66, 61, 54, 48, 80, 79, 71, 59, 19, 22,
       77, 58, 17, 72, 70, 18, 21, 20, 68, 74, 76, 75, 84, 85, 78, 15, 82,
       95, 13, 83, 81, 87, 88, 16, 91, 92, 97, 89, 90, 86, 94, 14, 99])

In [19]:
datos.Edad.max()

99

In [20]:
datos['Edad'].max()

99

In [21]:
datos['Años de Estudio'].max()

17

In [22]:
datos.Años de Estudio.max()

SyntaxError: invalid syntax (ipython-input-22-2344663015.py, line 1)

In [ ]:
print(f"La edad mínima es {datos.Edad.min()} y la edad máxima es {datos.Edad.max()}")

In [ ]:
print(f"La edad mínima es {datos.Edad.min()} y la edad máxima es {datos.Edad.max()}")

#### <font color='red'>Observación</font>
***
> La variable edad puede ser clasificada de tres formas diferentes:
> 1. <b>CUANTITATIVA DISCRETA</b> - cuando representa años completos (números enteros);
> 2. <b>CUANTITATIVA CONTINUA</b> - cuando representa la edad exacta, siendo representada por fracciones de años;
> 3. <b>CUALITATIVA ORDINAL</b> - cuando representa  intervalos de edad.

### Variables cuantitativas continuas

► Variables que representan un conteo o medición que asume valores en escalas continuas (números reales).

In [ ]:
print(f"La Altura mínima es {datos['Altura'].min()} y la altura máxima es {datos['Altura'].max()}")

In [ ]:
print(f"La altura mínima es:{datos['Altura'].min()} y la altura máxima es:{datos['Altura'].max()}")

# <font color=#56B4E9>2 DISTRIBUCIÓN DE FRECUENCIAS</font>
***

El primer paso en un análisis es conocer el comportamiento de las variables involucradas en el estudio. Utilizando técnicas estadísticas como el análisis de las <b> DISTRIBUCIONES DE FRECUENCIAS </b> e <b> HISTOGRAMAS </b> podemos evaluar mejor cómo se distribuyen los fenómenos del estudio.



## <font color=#56B4E9>2.1 Distribución de frecuencias para variables cualitativas</font>
***

### Método 1

https://pandas.pydata.org/pandas-docs/version/0.22/generated/pandas.Series.value_counts.html

In [ ]:
datos.Sexo.value_counts()

In [ ]:
datos.Sexo.value_counts(normalize=True)

In [ ]:
datos.Sexo.value_counts(normalize=True)*100


In [ ]:
datos.Sexo.value_counts(normalize=True).round(4)*100

In [ ]:
frecuencia_sexo=datos.Sexo.value_counts()
frecuencia_sexo

In [ ]:
porcentaje_sexo=datos.Sexo.value_counts(normalize=True).round(4)*100
porcentaje_sexo

In [ ]:
pd.DataFrame({'Frecuencia':frecuencia_sexo,'Porcentaje %':porcentaje_sexo})

In [ ]:
dist_frec_cualitativa=pd.DataFrame({'Frecuencia':frecuencia_sexo,'Porcentaje %':porcentaje_sexo})

In [ ]:
dist_frec_cualitativa

In [ ]:
dist_frec_cualitativa.rename(index={0:'Masculino',1:'Femenino'},inplace=True)

In [ ]:
dist_frec_cualitativa

### Método 2

https://pandas.pydata.org/pandas-docs/version/0.22/generated/pandas.crosstab.html

In [ ]:
sexo = {0: 'Masculino',
        1: 'Femenino'}

color = {0: 'Indigena',
         2: 'Blanco',
         4: 'Negro',
         6: 'Amarillo',
         8: 'Pardo',
         9: 'Sin declarar'}

In [ ]:
frecuencia=pd.crosstab(datos.Sexo,datos.Color)
frecuencia

In [ ]:
frecuencia.rename(index=sexo,inplace=True)
frecuencia.rename(columns=color,inplace=True)
frecuencia

In [ ]:
porcentaje_sexo=datos.Sexo.value_counts(normalize=True).round(4)*100
porcentaje_color=datos.Color.value_counts(normalize=True).round(4)*100

In [ ]:
porcentaje=pd.crosstab(datos.Sexo,datos.Color,normalize=True)*100
porcentaje.rename(index=sexo,inplace=True)
porcentaje.rename(columns=color,inplace=True)
porcentaje

## <font color=#56B4E9>2.2 Distribución de frecuencia para variables cuantitativas (clases personalizadas)</font>
***

### Paso 1 - Especifique los límites de cada clase

Utilice la siguiente clasificación:

<b>A</b> ► Más de 20 SM

<b>B</b> ► De 10 a 20 SM

<b>C</b> ► De 4 a 10 SM

<b>D</b> ► De 2 a 4 SM

<b>E</b> ► Hasta 2 SM

donde <b>SM</b> es el valor del salario mínimo en ese momento. En nuestro caso <b>788 mil pesos colombianos</b> (2018):

<b>A</b> ► Más de 15.760

<b>B</b> ► De 7.880 a 15.760

<b>C</b> ► De 3.152 a 7.880

<b>D</b> ► De 1.576 a 3.152

<b>E</b> ► Hasta 1.576


In [ ]:
datos.Ingreso.min()

In [ ]:
datos.Ingreso.max()

In [ ]:
clases=[0,1576,3152,7880,15760,200000]

In [ ]:
label=['E','D','C','B','A']

### Paso 2 - Crear la tabla de frecuencias

https://pandas.pydata.org/pandas-docs/version/0.22/generated/pandas.cut.html

In [ ]:
frecuencia=pd.cut(x=datos.Ingreso,bins=clases,labels=label,include_lowest=True)

In [ ]:
frecuencia=pd.value_counts(pd.cut(x=datos.Ingreso,bins=clases,labels=label,include_lowest=True))
frecuencia

In [ ]:
porcentaje=pd.value_counts(pd.cut(x=datos.Ingreso,bins=clases,labels=label,include_lowest=True),normalize=True)*100
porcentaje

In [ ]:
porcentaje=pd.value_counts(pd.cut(x=datos.Ingreso,bins=clases,labels=label,include_lowest=True),normalize=True).round(4)*100
porcentaje

In [ ]:
dist_frec_cuantitativa_personalizada=pd.DataFrame({'Frecuencia':frecuencia,'Porcentaje %':porcentaje})
dist_frec_cuantitativa_personalizada

In [ ]:
dist_frec_cuantitativa_personalizada.sort_index(ascending=False)

## <font color=#56B4E9>2.3 Distribución de frecuencia para variables cuantitativas (clases de amplitud fija)</font>
***

### Importando bibliotecas

http://www.numpy.org/

In [32]:
import numpy as np


### Paso 1 - Difiniendo el número de clases

#### Regla de Sturges

# $$k = 1 + \frac {10}{3}\log_{10}n$$

In [33]:
datos.shape

(76840, 7)

In [34]:
n=datos.shape[0]
n

76840

In [35]:
k=1+((10/3)*np.log10(n))
k

np.float64(17.285291187298853)

### Paso 2 - Crear la tabla de frecuencias

In [36]:
frecuencia_sturges=pd.value_counts(pd.cut(x=datos.Ingreso,bins=17,include_lowest=True))
frecuencia_sturges

/tmp/ipython-input-36-3468901511.py:1: FutureWarning: pandas.value_counts is deprecated and will be removed in a future version. Use pd.Series(obj).value_counts() instead.
  frecuencia_sturges=pd.value_counts(pd.cut(x=datos.Ingreso,bins=17,include_lowest=True))


Ingreso
(-200.001, 11764.706]       75594
(11764.706, 23529.412]       1022
(23529.412, 35294.118]        169
(35294.118, 47058.824]         19
(47058.824, 58823.529]         16
(94117.647, 105882.353]         6
(58823.529, 70588.235]          5
(70588.235, 82352.941]          4
(188235.294, 200000.0]          3
(82352.941, 94117.647]          1
(117647.059, 129411.765]        1
(129411.765, 141176.471]        0
(105882.353, 117647.059]        0
(141176.471, 152941.176]        0
(152941.176, 164705.882]        0
(164705.882, 176470.588]        0
(176470.588, 188235.294]        0
Name: count, dtype: int64

In [37]:
porcentaje_sturges=pd.value_counts(pd.cut(x=datos.Ingreso,bins=17,include_lowest=True),sort=False,normalize=True)*100
porcentaje_sturges

/tmp/ipython-input-37-2414209869.py:1: FutureWarning: pandas.value_counts is deprecated and will be removed in a future version. Use pd.Series(obj).value_counts() instead.
  porcentaje_sturges=pd.value_counts(pd.cut(x=datos.Ingreso,bins=17,include_lowest=True),sort=False,normalize=True)*100


Ingreso
(-200.001, 11764.706]       98.378449
(11764.706, 23529.412]       1.330036
(23529.412, 35294.118]       0.219938
(35294.118, 47058.824]       0.024727
(47058.824, 58823.529]       0.020822
(58823.529, 70588.235]       0.006507
(70588.235, 82352.941]       0.005206
(82352.941, 94117.647]       0.001301
(94117.647, 105882.353]      0.007808
(105882.353, 117647.059]     0.000000
(117647.059, 129411.765]     0.001301
(129411.765, 141176.471]     0.000000
(141176.471, 152941.176]     0.000000
(152941.176, 164705.882]     0.000000
(164705.882, 176470.588]     0.000000
(176470.588, 188235.294]     0.000000
(188235.294, 200000.0]       0.003904
Name: proportion, dtype: float64

In [38]:
dist_frec_cuantitativa_amplitud_fija=pd.DataFrame({'Frecuencia':frecuencia_sturges,'Porcentaje %':porcentaje_sturges})
dist_frec_cuantitativa_amplitud_fija

Frecuencia  Porcentaje %
Ingreso                                           
(-200.001, 11764.706]          75594     98.378449
(11764.706, 23529.412]          1022      1.330036
(23529.412, 35294.118]           169      0.219938
(35294.118, 47058.824]            19      0.024727
(47058.824, 58823.529]            16      0.020822
(58823.529, 70588.235]             5      0.006507
(70588.235, 82352.941]             4      0.005206
(82352.941, 94117.647]             1      0.001301
(94117.647, 105882.353]            6      0.007808
(105882.353, 117647.059]           0      0.000000
(117647.059, 129411.765]           1      0.001301
(129411.765, 141176.471]           0      0.000000
(141176.471, 152941.176]           0      0.000000
(152941.176, 164705.882]           0      0.000000
(164705.882, 176470.588]           0      0.000000
(176470.588, 188235.294]           0      0.000000
(188235.294, 200000.0]             3      0.003904

## <font color=#56B4E9>2.4 Histograma</font>
***

El <b>HISTOGRAMA</b> es la representación gráfica de una distribución de frecuencia. Es un gráfico formado por un conjunto de rectángulos colocados uno al lado del otro, donde el área de cada rectángulo es proporcional a la frecuencia de la clase que representa.

### Importando la biblioteca

https://seaborn.pydata.org/

In [ ]:
import seaborn as sns

In [ ]:
ax=sns.distplot(datos.Altura,kde=False)
ax.figure.set_size_inches(12,6)
ax.set_title('Distribución de frecuencias - Altura',fontsize=18)
ax.set_xlabel('Altura (metros)',fontsize=14)
ax

In [ ]:
ax=sns.histplot(datos.Altura,kde=True)
ax.figure.set_size_inches(12,6)
ax.set_title('Distribución de frecuencias - Altura -KDE',fontsize=18)
ax.set_xlabel('Altura (metros)',fontsize=14)
ax

# <font color=#56B4E9>3 MEDIDAS DE TENDENCIA CENTRAL</font>
***

## DataFrame de ejemplo

In [39]:
df = pd.DataFrame(data = {'María': [8, 10, 4, 8, 6, 10, 8],
                          'Pedro': [10, 2, 0.5, 1, 3, 9.5, 10],
                          'Pablo': [7.5, 8, 7, 8, 8, 8.5, 7]},
                  index = ['Matemática',
                           'Portugués',
                           'Inglés',
                           'Geografía',
                           'Historia',
                           'Física',
                           'Química'])
df.rename_axis('Asignaturas', axis = 'columns', inplace = True)
df

Asignaturas  María  Pedro  Pablo
Matemática       8   10.0    7.5
Portugués       10    2.0    8.0
Inglés           4    0.5    7.0
Geografía        8    1.0    8.0
Historia         6    3.0    8.0
Física          10    9.5    8.5
Química          8   10.0    7.0

## <font color=#56B4E9>3.1 Media aritmética</font>
***

Es representada por $\mu$ cuando se refiere a la población y por $\bar{X}$ cuando se refiere a la muestra

# $$\mu = \frac 1n\sum_{i=1}^{n}X_i$$

donde

$n$ = número de observaciones (registros)

$X_i$ = valor de la i-ésima observación (registro)

In [40]:
(8+10+4+8+6+10+8)/7

7.714285714285714

In [41]:
df['María'].mean()

np.float64(7.714285714285714)

In [42]:
datos.groupby(['Sexo'])['Ingreso'].mean()

Sexo
0    2192.441596
1    1566.847393
Name: Ingreso, dtype: float64

## <font color=#56B4E9>3.2 Mediana</font>
***

Para obtener la mediana de un conjunto de datos, debemos proceder de la siguiente manera:
1. Ordenar el conjunto de datos;
2. Identificar el número de observaciones (registros) del conjunto de datos ($n$);
3. Identicar el elemento del medio:

> Cuando $n$ sea impar, la posición del elemento del medio se obtendrá de la siguiente manera:


# $$Elemento_{Md} = \frac{n+1}{2}$$

> Cuando $n$ sea par, la posición del elemento mediano se obtendrá de la siguiente manera:


# $$Elemento_{Md} = \frac{n}2$$

4. Obtener la mediana:

> Cuando $n$ sea impar:


# $$Md = X_{Elemento_{Md}}$$

> Cuando $n$ sea par:


# $$Md = \frac{X_{Elemento_{Md}} + X_{Elemento_{Md}+1}}2$$
***

### Ejemplo 1 - n impar

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img002.png' width='40%' style="float:left">

In [ ]:
notas_maria=df['María']
notas_maria

In [ ]:
notas_maria=notas_maria.sort_values()
notas_maria

In [ ]:
n=notas_maria.shape[0]
n

In [ ]:
notas_maria=notas_maria.reset_index()
notas_maria

In [ ]:
elemento_md=(n+1)/2
elemento_md


In [ ]:
notas_maria.loc[elemento_md-1]

In [ ]:
notas_maria['María'].median()

### Ejemplo 2 - n par

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img003.png' width='50%' style="float:left">

In [ ]:
notas_pedro=df['Pedro'].sample(6,random_state=200)
notas_pedro

In [ ]:
notas_pedro=notas_pedro.sort_values()
notas_pedro

In [ ]:
n=notas_pedro.shape[0]
n

In [ ]:
notas_pedro=notas_pedro.reset_index()
notas_pedro

In [ ]:
elemento_md=n/2
elemento_md

In [ ]:
(notas_pedro.loc[elemento_md-1][1]+notas_pedro.loc[elemento_md][1])/2

In [ ]:
notas_pedro['Pedro'].median()

### Obtener la mediana en nuestro conjunto de datos

## <font color=#56B4E9>3.3 Moda</font>
***

La moda se puede definir como el valor más frecuente de un conjunto de datos. La moda es ampliamente utilizada para datos cualitativos.

In [43]:
df

Asignaturas  María  Pedro  Pablo
Matemática       8   10.0    7.5
Portugués       10    2.0    8.0
Inglés           4    0.5    7.0
Geografía        8    1.0    8.0
Historia         6    3.0    8.0
Física          10    9.5    8.5
Química          8   10.0    7.0

In [44]:
df.mode()

Asignaturas  María  Pedro  Pablo
0                8   10.0    8.0

### Obteniendo la moda de nuestro dataset

In [ ]:
df.mode()


In [25]:
ejemplo=pd.Series([1,2,2,3,4,4,5,6,7])
ejemplo

0    1
1    2
2    2
3    3
4    4
5    4
6    5
7    6
8    7
dtype: int64

In [45]:
ejemplo.mode()

0    2
1    4
dtype: int64

In [23]:
datos.Ingreso.mode()


0    788
Name: Ingreso, dtype: int64

In [46]:
datos.Altura.mode()

0    1.568128
1    1.671225
2    1.681659
3    1.692977
4    1.708163
5    1.708370
6    1.753842
7    1.779073
8    1.796462
Name: Altura, dtype: float64

## <font color=#56B4E9>3.4 Relación entre media, mediana e moda</font>
***

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img004.png' width='80%'>

### Evaluando la variable INGRESO

***

### Evaluando la variable ALTURA

***

### Evaluando la variable AÑOS DE ESTUDIO

# <font color=#56B4E9>4 MEDIDAS DE LOCALIZACIÓN</font>
***

## <font color=#56B4E9>4.1 Cuartiles, deciles y percentiles</font>
***

Hay una serie de medidas de posición similares en su diseño a la mediana, aunque no son medidas de tendencia central. Como se sabe, la mediana divide la distribución en dos partes iguales en términos del número de elementos en cada parte. Los cuartiles, por otro lado, permiten que la distribución se divida en cuatro partes iguales en términos del número de elementos en cada uno; deciles en diez partes y centiles en cien partes iguales.

## <font color=#56B4E9>4.2 Box-plot</font>
***

El *box plot* da una idea de la posición, dispersión, asimetría, colas y valores extremos. La posición central está dada por la mediana y la dispersión por $IIQ$ (distancia inter cuartil). Las posiciones relativas de $Q1$, $Mediana$ y $Q3$ dan una idea de la simetría de la distribución. Las longitudes de las colas están dadas por las líneas que van desde el rectángulo a los valores remotos y por los valores extremos.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img005.png' width='65%'>

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1752-estadistica-con-python-parte1/img006.png' width='80%'>

# <font color=#56B4E9>5 MEDIDAS DE DISPERSIÓN</font>
***

Aunque las medidas de tendencia central y de localización proporcionan un resumen muy importante de los datos, pueden no ser suficientes para caracterizar diferentes conjuntos, especialmente cuando las observaciones de una distribución dada presentan datos muy dispersos.

## <font color=#56B4E9>5.1 Desviación media absoluta</font>
***


# $$DM = \frac 1n\sum_{i=1}^{n}|X_i-\bar{X}|$$


## <font color=#56B4E9>5.2 Varianza</font>
***

### Varianza

La varianza se construye a partir de las diferencias entre cada observación y la media de los datos, es decir, la desviación alrededor de la media. Al calcular la varianza, las desviaciones alrededor de la media son elevadas al cuadrado.

### Varianza de la población

# $$\sigma^2 = \frac 1n\sum_{i=1}^{n}(X_i-\mu)^2$$

### Varianza de la muestra

# $$S^2 = \frac 1{n-1}\sum_{i=1}^{n}(X_i-\bar{X})^2$$

## <font color=#56B4E9>5.3 Desviación estándar</font>
***

Una de las restricciones de la varianza es el hecho de que proporciona medidas cuadráticas de las unidades originales; la varianza de las medidas de longitud, por ejemplo, está en unidades de área. Por lo tanto, el hecho de que las unidades sean diferentes dificulta la comparación de la dispersión con las variables que la definen. Una forma de eliminar esta dificultad es considerar su raíz cuadrada.

### Desviación estándar de la población

# $$\sigma = \sqrt{\frac 1n\sum_{i=1}^{n}(X_i-\mu)^2} \Longrightarrow \sigma = \sqrt{\sigma^2}$$


### Desviación estándar de la muestra

# $$S = \sqrt{\frac 1{n-1}\sum_{i=1}^{n}(X_i-\bar{X})^2} \Longrightarrow S = \sqrt{S^2}$$